activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -89,6 +89,7 @@ def update_counters_via_scope(klass, foreign_key, by)
         end
 
         def find_target?
+          raise if @owner.readonly?
```

Comment:
Should we raise a specific exception here to tell users what this means? BTW, why `readonly?` should it not be `strict?`

Reply:
I think this line can be removed actually (cc @eileencodes).  But yes we should add an exception that is specific to this.